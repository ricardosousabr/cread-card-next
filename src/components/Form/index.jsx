import {
  ContainerForm,
  InputLarge,
  BoxInputSmall,
  InputSmall,
  Text,
  BoxButton,
} from './styles'

export default function Form() {
  return (
    <div>
      <div>
        <ContainerForm action="">
          <div>
            <Text>Número do cartão</Text>
            <InputLarge type="number" />
          </div>
          <div>
            <Text>Nome do titular</Text>
            <InputLarge type="number" />
          </div>
          <BoxInputSmall>
            <div>
              <Text>Expiração</Text>
              <InputSmall type="number" />
            </div>
            <div>
              <Text>CVC</Text>
              <InputSmall type="number" />
            </div>
          </BoxInputSmall>
          <BoxButton>
            <button>Adicionar</button>
          </BoxButton>
        </ContainerForm>
      </div>
    </div>
  )
}
