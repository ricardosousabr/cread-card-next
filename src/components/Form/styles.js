import styled from 'styled-components'

export const ContainerForm = styled.form`
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  height: 250px;
`

export const InputLarge = styled.input`
  width: 200px;
  height: 40px;
  background: black;
  border: none;
  border-radius: 5px;
  color: white;
`

export const BoxInputSmall = styled.div`
  display: flex;
  justify-content: space-between;
`

export const InputSmall = styled.input`
  width: 90px;
  height: 40px;
  background: black;
  border: none;
  border-radius: 5px;
  color: white;
`

export const Text = styled.p`
  color: white;
  font-size: 14px;
  padding: 5px;
`

export const BoxButton = styled.div`
  margin-top: 10px;
`
