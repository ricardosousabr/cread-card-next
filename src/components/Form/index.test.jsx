/**
 * @jest-environment jsdom
 */

import React from 'react'
import Form from './index'
import renderer from 'react-test-renderer'

test('HomePage snapshot', () => {
  const component = renderer.create(<Form />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
