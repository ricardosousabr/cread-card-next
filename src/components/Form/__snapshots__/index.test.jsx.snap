// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage snapshot 1`] = `
<div>
  <div>
    <form
      action=""
      className="styles__ContainerForm-sc-wedmof-0 dPZUAw"
    >
      <div>
        <p
          className="styles__Text-sc-wedmof-4 iCLgfV"
        >
          Número do cartão
        </p>
        <input
          className="styles__InputLarge-sc-wedmof-1 gGEqVy"
          type="number"
        />
      </div>
      <div>
        <p
          className="styles__Text-sc-wedmof-4 iCLgfV"
        >
          Nome do titular
        </p>
        <input
          className="styles__InputLarge-sc-wedmof-1 gGEqVy"
          type="number"
        />
      </div>
      <div
        className="styles__BoxInputSmall-sc-wedmof-2 lnSgfT"
      >
        <div>
          <p
            className="styles__Text-sc-wedmof-4 iCLgfV"
          >
            Expiração
          </p>
          <input
            className="styles__InputSmall-sc-wedmof-3 kJOAep"
            type="number"
          />
        </div>
        <div>
          <p
            className="styles__Text-sc-wedmof-4 iCLgfV"
          >
            CVC
          </p>
          <input
            className="styles__InputSmall-sc-wedmof-3 kJOAep"
            type="number"
          />
        </div>
      </div>
      <div
        className="styles__BoxButton-sc-wedmof-5 jprGjP"
      >
        <button
          className="styles__ButtonAddCard-sc-11nj3cj-0 eZHHSY"
        >
          Adicionar
        </button>
      </div>
    </form>
  </div>
</div>
`;
